### PR TITLE
Fix a few issues on study manager

### LINKED ISSUE #374: Issues on storing studying in Study Manager

### PR DESCRIPTION
**Pull request by IsaacYangSLA**
Fixes #374

### CODE REVIEW

File: nvflare/apis/impl/study_manager.py [hunk 1]
```diff
@@ -94,14 +126,18 @@ def list_active_studies(self) -> List[str]:
...
 
         """
-        now = datetime.datetime.utcnow()
+        today = date.today()
         active_studies = list()
-        for k, v in self._existing_studies:
-            if v["start_date"] >= now and v["end_date"] <= now:
-                active_studies.append(k)
+        for st in self._existing_studies:
+            meta = self._storage.get_meta(st)
```
Comment (YuanTingHsieh) [inline nvflare/apis/impl/study_manager.py:132]: Will this get_meta raise Exception?
Should we handle that?

File: nvflare/apis/impl/study_manager.py [hunk 2]
```diff
@@ -33,12 +33,18 @@ def custom_json_encoder(obj):
...
         self._storage = storage
-        self._tmp_dir = tempfile.TemporaryDirectory()
-        self._existing_studies = dict()
+        try:
+            self._existing_studies = json.loads(self._storage.get_data(StudyManager.STORAGE_KEY).decode("utf-8"))
+        except BaseException:
+            self._existing_studies = list()
 
-    def create_study(self, study: Study) -> Tuple[Optional[Study], str]:
+    def add_study(self, study: Study) -> Tuple[Optional[Study], str]:
```
Comment (YuanTingHsieh) [inline nvflare/apis/impl/study_manager.py:47]: maybe we want to also update the docstrings.

Other than this,
For the method docstrings after Returns: there should be a new line.

File: nvflare/apis/impl/study_manager.py [hunk 3]
```diff
@@ -54,7 +60,7 @@ def create_study(self, study: Study) -> Tuple[Optional[Study], str]:
         Returns: updated study info (e.g. create_time is set)
 
         """
-        if self.get_study(study.name):
+        if study.name in self._existing_studies:
             return None, "Unable to create duplicated study name."
         if not study.participating_clients:
             return None, "Study has no participating clients."
```
Comment (YuanTingHsieh) [inline nvflare/apis/impl/study_manager.py:66]: if we change the method name, maybe change this error msg as well -> `Unable to add duplicated study name.` ?

File: nvflare/apis/impl/study_manager.py (hunk 2 above)
Comment (YuanTingHsieh) [reply nvflare/apis/impl/study_manager.py:47]: Or should we change the spec name from "add" to "create"?

File: nvflare/apis/impl/study_manager.py [hunk 4]
```diff
@@ -66,17 +72,43 @@ def create_study(self, study: Study) -> Tuple[Optional[Study], str]:
...
+        serialized_study = self._study_to_bytes(study)
+        self._storage.create_object(
             study.name, serialized_study, {"start_date": study.start_date, "end_date": study.end_date}
         )
-        deserialized_study = json.loads(serialized_study.decode("utf-8"))
-        deserialized_study["start_date"] = datetime.date.fromisoformat(deserialized_study["start_date"])
-        deserialized_study["end_date"] = datetime.date.fromisoformat(deserialized_study["end_date"])
-        stored_study = Study(**deserialized_study)  # for debugging purpose
+        self._existing_studies.append(study.name)
+        self._storage.create_object(
```
Comment (YuanTingHsieh) [inline nvflare/apis/impl/study_manager.py:86]: should we have try block around this?

File: nvflare/apis/impl/study_manager.py (hunk 1 above)
Comment (IsaacYangSLA) [reply nvflare/apis/impl/study_manager.py]: Yes.  When it raises exception, that study should not be included.

File: nvflare/apis/impl/study_manager.py (hunk 4 above)
Comment (IsaacYangSLA) [reply nvflare/apis/impl/study_manager.py]: Right.  Fixed.

**Review by YuanTingHsieh — APPROVED** [review]
LGTM

**Pull request merged**